lib/recipe/widgets/ingredient_form.dart

```diff
@@ -39,102 +39,138 @@ class IngredientForm extends StatelessWidget {
+                              ingredientDescription: value,
+                              recipe: state.recipes[recipeIndex]));
+                    },
+                    textCapitalization: TextCapitalization.sentences,
+                    textInputAction: TextInputAction.next,
+                  ),
+                  DropdownSearch<String>(
+                      dropdownSearchDecoration: const InputDecoration(
+                        icon: Icon(Icons.food_bank_rounded),
+                        labelText: "Cooking state",
+                        helperText: 'How the ingredient is prepared',
+                      ),
+                      mode: Mode.MENU,
+                      showSelectedItems: true,
+                      showSearchBox: true,
+                      items: cookingStates,
+                      onChanged: (String? answer) => context
+                          .read<RecipeBloc>()
+                          .add(IngredientCookingStateChanged(
+                              ingredient: state.recipes[recipeIndex]
+                                  .ingredients[ingredientIndex],
+                              cookingState: answer!,
+                              recipe: state.recipes[recipeIndex])),
+                      selectedItem: state.recipes[recipeIndex]
+                          .ingredients[ingredientIndex].cookingState),
+                  Visibility(
+                    visible: (state.recipes[recipeIndex]
+                            .ingredients[ingredientIndex].cookingState ==
+                        "Other (please specify)"),
+                    child: TextFormField(
+                      initialValue: state.recipes[recipeIndex]
+                          .ingredients[ingredientIndex].customCookingState,
+                      decoration: InputDecoration(
+                        icon: const Icon(Icons.set_meal_rounded),
+                        labelText: 'Specify cooking state',
+                        helperText: 'Cooking state e.g. Barbequed',
+                        errorText: (state.recipes[recipeIndex]
+                                    .ingredients[ingredientIndex].customName ==
+                                null)
+                            ? 'Enter a cooking state e.g. Barbequed'
+                            : null,
+                      ),
+                      onChanged: (value) {
+                        context.read<RecipeBloc>().add(
+                            IngredientCustomCookingStateChanged(
+                                ingredient: state.recipes[recipeIndex]
+                                    .ingredients[ingredientIndex],
+                                customCookingState: value,
+                                recipe: state.recipes[recipeIndex]));
+                      },
+                      textInputAction: TextInputAction.next,
+                      textCapitalization: TextCapitalization.sentences,
```

Comment:
This is very nice for the "Other" cooking state, but I'm getting bottom overflows when trying to fill this (see below). I'd guess that something somewhere might need to be wrapped in an Expanded widget or similar, as I remember dealing with similar issue in collection. LMK if you can reproduce.

<img width="380" alt="b2" src="https://user-images.githubusercontent.com/45038919/168496195-c176338c-7b33-4ad3-8928-108083875b42.png">